backend/src/test/java/apsi/team3/backend/services/EventServiceTest.java

```diff
@@ -0,0 +1,127 @@
+        assertEquals(Optional.of(eventDTO), result);
+    }
+
+    @Test
+    public void testGetAllEvents() {
+        List<Event> eventList = new ArrayList<>();
+        User organizer = new User(1L, "login", "hash", "salt", UserType.ORGANIZER, null);
+        eventList.add(new Event(1L, "1", null, null, null, null, "desc", organizer, null, null));
+        eventList.add(new Event(2L, "2", null, null, null, null, "desc", organizer, null, null));
+        eventList.add(new Event(3L, "3", null, null, null, null, "desc", organizer, null, null));
+        when(eventRepository.findAll()).thenReturn(eventList);
+        List<EventDTO> eventDTOList = new ArrayList<>();
+        for (var event: eventList) {
+            eventDTOList.add(DTOMapper.toDTO(event));
+        }
+        List<EventDTO> result = eventService.getAllEvents();
+        assertEquals(eventDTOList, result);
+    }
+
+    @Test
+    public void testCreateNull() {
+        EventDTO nullEventDto = new EventDTO(null, null, null, null, null, null, null, null, null);
+        assertThrows(ApsiValidationException.class, () -> eventService.create(nullEventDto));
+    }
+
+    @Test
+    public void testCreate() {
+        EventDTO eventDTO = new EventDTO(null, "name", null, null, null, null, "desc", 1L, null);
+        Event event = DTOMapper.toEntity(eventDTO);
+        try (var securityContextHolderMockedStatic = mockStatic(SecurityContextHolder.class)) {
+            User user = new User(420L, "login", "hash", "salt", UserType.ORGANIZER, null);
+            SecurityContext securityContextMock = mock(SecurityContext.class);
+            securityContextHolderMockedStatic.when(SecurityContextHolder::getContext).thenReturn(securityContextMock);
+            Authentication authenticationMock = mock(Authentication.class);
+            when(securityContextMock.getAuthentication()).thenReturn(authenticationMock);
+            when(authenticationMock.getPrincipal()).thenReturn(user);
+            when(eventRepository.save(any())).thenReturn(event);
+            EventDTO result = eventService.create(eventDTO);
+            event.setOrganizer(user);
+            verify(eventRepository).save(ArgumentMatchers.refEq(event));
+            assertEquals(result, eventDTO);
+        } catch (Exception e) {
+            fail();
+        }
+    }
+
+    @Test
+    public void testReplace() {
+        EventDTO eventDTO = new EventDTO(null, "name", LocalDate.of(2024, 4, 27), null, null, null, "desc", 1L, null);
+        when(eventRepository.save(any())).thenReturn(DTOMapper.toEntity(eventDTO));
+        try {
+            assertEquals(eventService.replace(eventDTO), eventDTO);
+        } catch (Exception e) {
+            fail();
+        }
+    }
+
+    @Test
+    public void testDelete() {
+        Long idToDelete = 18L;  // "How to forget?" #pdk
+        eventService.delete(idToDelete);
+        verify(eventRepository).deleteById(idToDelete);
+    }
+
+    @Test
+    public void testNotExist() {
```

Comment:
ten test jest raczej do wywalenia - sprawdzasz bardziej prawidłowe działanie biblioteki do mockowania niż nasz kod (a sprawdzenie, czy event istnieje i tak idzie z klasy `JpaRepository` a nie z naszej implementacji

Reply:
napisanie tego testu miało niewielki sens, ale usuwanie go również imo nie jest konieczne. Zawsze to odrobinę większe pokrycie kodu testami (chyba)